### ISSUE TITLE
Newlines not supported in :has-text selectors

### ISSUE THREAD

**Issue by astearns** [microsoft/playwright#17604]
**Context:**
- Playwright Version: 1.20.0
- Operating System: Mac
- Node.js version: 14.18.1
- Browser: Chromium
- Extra: 

**Code Snippet**

I have some code that selects a specific item from a list that depends on text content. In some tests, this text content contains a newline character. This fails with an unsupported token error

```javascript
// comment is a string passed in to a function that creates this:
const specificNameLocator = `.comment-content-wrapper:has-text("${comment}") .user-name`;
// then uses that locator to click
```

**Describe the bug**

When `string` contains a newline character, I get this error:

Unsupported token "BADSTRING" while parsing selector ".comment-content-wrapper:has-text("replied6619+6710
next line") .user-name"

**Comment by dgozman** [comment]
@astearns `:has-text` is a part of a CSS selector, so you have to provide a valid CSS selector. For example, to properly escape `\n` in CSS, you should do `\\a ` - note the space at the end. More generically, you'd want to call [`CSS.escape()`](https://developer.mozilla.org/en-US/docs/Web/API/CSS/escape) on the text you insert into `:has-text`.

An easier solution would be to replace any whitespace with a single space, because Playwright normalizes whitespace anyway:

```js
const specificNameLocator = `.comment-content-wrapper:has-text("${comment.replace(/\s+/g, ' ')}") .user-name`;
```

---

That said, I'd suggest you to use locators and [pass `hasText` option](https://playwright.dev/docs/api/class-locator#locator-filter-option-has-text), so that Playwright handles it for you. https://github.com/microsoft/playwright/pull/17798 fixed an issue related to that, and should be available with v1.27 release. Let me know if that helps.

**Comment by dgozman** [comment]
Fixed by #17798.
